src/gridiron/controls/Control.cpp

```diff
@@ -25,26 +25,22 @@
 #include <gridiron/controls/Control.hpp>
 
 namespace GridIron {
+    class Page;
 
     Control::Control(const char *id, std::shared_ptr<Control> parent) :
         This{shared_from_this()},
```

Comment:
I'm afraid `shared_from_this()` can't be called from within the constructor, because the `shared_ptr` doesn't exist yet!  Also if you have an object with a `shared_ptr` to itself, it will never get destructed!  You should remove the `This` member and add a member function that returns `shared_from_this()` (and make sure not call it in the constructor).

Reply:
Good call. Thanks for finding this!
The This member is a property which is aimed at making the access semantics pretty. I think I can make a shared from this property class which takes care of it....

Reply:
Fixed in latest.

Reply:
Thank you!